exchange/utils.go

```diff
@@ -998,20 +966,27 @@ func applyFPD(fpd map[openrtb_ext.BidderName]*firstpartydata.ResolvedFirstPartyD
 	}
 
 	if fpdToApply.Site != nil {
-		bidRequest.Site = fpdToApply.Site
+		reqWrapper.Site = fpdToApply.Site
 	}
 
 	if fpdToApply.App != nil {
-		bidRequest.App = fpdToApply.App
+		reqWrapper.App = fpdToApply.App
 	}
 
 	if fpdToApply.User != nil {
-		//BuyerUID is a value obtained between fpd extraction and fpd application.
-		//BuyerUID needs to be set back to fpd before applying this fpd to final bidder request
-		if bidRequest.User != nil && len(bidRequest.User.BuyerUID) > 0 {
-			fpdToApply.User.BuyerUID = bidRequest.User.BuyerUID
+		if reqWrapper.User != nil {
+			if len(reqWrapper.User.BuyerUID) > 0 {
+				//BuyerUID is a value obtained between fpd extraction and fpd application.
+				//BuyerUID needs to be set back to fpd before applying this fpd to final bidder request
+				fpdToApply.User.BuyerUID = reqWrapper.User.BuyerUID
+			}
+			if len(reqWrapper.User.EIDs) > 0 {
```

Comment:
I don't think we want the conditional. Maybe this is simply just the statement `fpdToApply.User.EIDs = reqWrapper.User.EIDs` so it always executes. It's possible that there were EIDs but they were all removed bringing the count to 0. Thoughts?

Reply:
Done

Reply:
Discussed offline. More changes are in progress